lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
> exclude NetCDF

If we go the deconstructed route then unless/until we have some use case to guide us/force our hand, I'm in favour of leaving it out.

That said, if we stick with the callback route then I'm also inclined to leave it out because it behaves differently!